### DISCUSSION TITLE
[Feature Request] Performance Profiling Tools

### DISCUSSION THREAD

**Discussion by kevinschaich** [jotaijs/jotai-devtools#136] [Ideas]
[React & Chrome devtools have a number of profiling tools](https://legacy.reactjs.org/blog/2018/09/10/introducing-the-react-profiler.html) to help debug performance issues, e.g. flamegraphs, render charts:

![React Profiler Filtering](https://github.com/jotaijs/jotai-devtools/assets/9244728/99d5ac29-bdbf-46cd-8075-0638ddbe1169)

While these are super helpful I realize they're a big undertaking. Even some preliminary bar charts or histograms would be helpful.

Render time is probably not the right metric for us here. Just to brainstorm a few:

* Number of set calls to an atom in the last X milliseconds (how many components are writing to this atom?)
* Data volume in the last X milliseconds (are we setting a HUGE object or binary data every time?)
* Diff volume in the last X milliseconds (are we setting one tiny property on a HUGE object every time?)
* Number of listeners/observer updates in the last X milliseconds (how many components are re-rendering as a result of this atom's change)?

Maybe these are not exactly right but it would be useful to have something.

**Comment by arjunvegda** [comment]
Thanks for reaching out! 

It's a good idea! These feel more like analytics rather than performance measurements (nevertheless, still interesting). Let's continue brainstorming and groom this further.
